app/assets/javascripts/matrix.js

```diff
@@ -11,11 +11,12 @@ class Matrix {
     }
 
     this.num_rows = this.rows.length
-    this.num_cols = this.rows[0].length
 
     if (this.col_names) {
+      this.num_cols = this.col_names.length
```

Comment:
What is this meant to fix? This also breaks line 17, which can never raise now that num_cols is set from col_names instead of from rows[0]...